main.go

```diff
@@ -0,0 +1,200 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"net/http"
+	"sort"
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+)
+
+type CincinnatiGraph struct {
+	Nodes []CincinnatiNode `json:"nodes"`
+}
+
+type CincinnatiNode struct {
+	Version  *semver.Version   `json:"version"`
+	Payload  string            `json:"payload"`
+	Metadata map[string]string `json:"metadata"`
+}
+
+type Release struct {
+	Version  *semver.Version
+	Channel  string
+	Arch     string
+	Payload  string
+	Metadata map[string]string
+}
+
+type VersionReleases map[string]Release
+
+type ReleasesByChannel map[string]VersionReleases
+
+// fetchGraph fetches the upgrade graph for a given channel and architecture.
+func fetchGraph(client *http.Client, channel, arch string) (*CincinnatiGraph, error) {
+	url := fmt.Sprintf("https://api.openshift.com/api/upgrades_info/graph?channel=%s&arch=%s", channel, arch)
```

Comment:
nit: `fmt.Sprintf` assumes that your callers are playing nice with their `channel` and `arch` inputs.  You can use [`net/url`'s `Values.Encode`][1] if you want to be more defensive.  [Here][2] is how the CVO handles this, in case that helps :)

[1]: https://pkg.go.dev/net/url#Values.Encode
[2]: https://github.com/openshift/cluster-version-operator/blob/4ec8c48667e960ffeff65f1e272a27e2dabc6261/pkg/cincinnati/cincinnati.go#L95-L103